0006-cmx2cdp-protocol/README.md

```diff
@@ -1,84 +1,92 @@
+Configuration Data Transport (CDT) protocol is designed to communicate endpoint configuration data between Kaa services.
 
-## Requirements and constraints
-### Problems and possible solutions
+In Kaa architecture, some services are designed to manage endpoint configuration in different ways — store it, process it, publish it, etc.
+One of the core features in such context is the ability to store and retrieve endpoint configurations.
 
-1. _Configuration push._
+For this purpose, the CDT protocol was designed.
+It defines the endpoint configuration data structure so that it's interpreted by all involved services in the same way.
 
-    Solutions:
-    - CDP should broadcast event that new configuration is available.
+In CDT lingo, storing an endpoint configuration to a service is called *configuration push*, while retrieving it from a service is called *configuration pull*.
 
-2. _Configuration delivery confirmation after configuration push._
+To effectively perform these activities, endpoint configuration data should be communicated in a format recognized by all the services involved.
+To achieve this, CDT protocol was designed based on the [3/Messaging IPC][3/MIPC] protocol.
 
-    Solutions:
-    - CMX should broadcast event that particular endpoint has updated its configuration to specific version.
-    >**NOTE:** In case of CMX configuration pull no delivery confirmation is required since endpoint can
-    recognise delivery fail and request its configuration again.
+## Language
 
-3. _It is possible that there are two CDP instances configured to work with different
-content types and one originator service instance. When originator service publishes config event
-message with certain content type it is consumed by both CDPs._
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 
-    Solutions:
-    - CDP should filter messages by their content types.
+The following terms and definitions are used in this RFC.
 
-4. _If CMX asks for endpoint configuration but there is no configuration currently available,
-CDP should somehow inform about it._
+- **Configuration push**: sending an endpoint configuration data to a service for further storage and/or processing.
+- **Configuration pull**: retrieving an endpoint configuration data from a service.
+- **Source service**: any service that initiates a configuration push/pull.
+- **Target service**: any service that a configuration push/pull is initiated towards.
 
-    Solutions:
-    - Use HTTP status codes and arbitrary reason phrases.
+## Requirements and constraints
 
+CDT requirements:
 
-#### Key concept
+- Once a new endpoint configuration is pushed to a service, this service must be able to broadcast an event over [NATS](http://nats.io/) about the new configuration availability.
```

Comment:
s/pushed to a service/provisioned into the provider

It becomes easier to explain things once you define these two roles.